Wrappers.h

```diff
@@ -0,0 +1,50 @@
+// struct Headphones;
+// struct OscillatorSection;
+// struct FilterSection;
+// struct Headset;
+// struct GeneratorChain;
+#pragma once
+
+#include "Headset.h"
+#include "GeneratorChain.h"
+//==================== WRAPPER CLASSES ====================
+
+struct HeadphonesWrapper
+{
+    HeadphonesWrapper(Headphones* ptr) : headphonesPtr (ptr) { }
+    ~HeadphonesWrapper() { delete headphonesPtr; }
+
+    Headphones* headphonesPtr = nullptr;
+};
+
+struct OscillatorSectionWrapper
+{
+    OscillatorSectionWrapper(OscillatorSection* ptr) : oscillatorSectionPtr (ptr) { }
+    ~OscillatorSectionWrapper() { delete oscillatorSectionPtr; }
+
+    OscillatorSection* oscillatorSectionPtr = nullptr;
+};
+
+struct FilterSectionWrapper
+{
+    FilterSectionWrapper(FilterSection* ptr) : filterSectionPtr (ptr) { }
+    ~FilterSectionWrapper() { delete filterSectionPtr; }
+
+    FilterSection* filterSectionPtr = nullptr;
+};
+
+struct HeadsetWrapper
+{
+    HeadsetWrapper(Headset* ptr) : headsetPtr (ptr) { }
+    ~HeadsetWrapper() { delete headsetPtr; }
+
+    Headset* headsetPtr = nullptr;
+};
+
+struct GeneratorChainWrapper
+{
+    GeneratorChainWrapper(GeneratorChain* ptr) : genChainPtr (ptr) { }
+    ~GeneratorChainWrapper() { delete genChainPtr; }
+
+    GeneratorChain* genChainPtr = nullptr;
+};
```

Comment:
```
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
        Chances are extremely high that you implemented your 
        wrapper class functions in-class, because that is what 
        everyone does after watching the video where I implemented 
        them in-class.
```
- [ ] revise